src/bootstrap/flags.rs

```diff
@@ -430,14 +423,11 @@ Arguments:
     This subcommand accepts a number of paths to directories to the crates
     and/or artifacts to compile. For example:
 
-        ./x.py check library/core
-        ./x.py check library/core library/proc_macro
+        ./x.py check library/std
 
-    If no arguments are passed then the complete artifacts are compiled: std, test, and rustc. Note
-    also that since we use `cargo check`, by default this will automatically enable incremental
-    compilation, so there's no need to pass it separately, though it won't hurt. We also completely
-    ignore the stage passed, as there's no way to compile in non-stage 0 without actually building
-    the compiler.",
+    If no arguments are passed then all artifacts are compiled. Incremental
+    compilation is used by default, so the `-i` option is unnecessary. The
+    `--stage` option is ignored and the stage 0 compiler is always used.",
```

Comment:
Most of these statements aren't true 🙃 check takes a --stage flag and incremental is not used by default. Saying "all artifacts are built" I guess gets the right idea across, but it's not very precise since you could interpret that to mean it builds everything before checking.

Reply:
I tried `x.py check --stage 1`. It went badly. I got this:
```
error[E0463]: can't find crate for `core`
error[E0463]: can't find crate for `compiler_builtins`
error[E0463]: can't find crate for `std`
```
Followed by a zillion compile errors.

`--stage` is an option that is present for every subcommand, but I think it's not useful and/or meaningful for a number of them: `check`, `clippy`, `fix`, `fmt`, `clean`, `setup`?

Reply:
`check --stage 1` used to work at some point, if it's broken now that's a regression. It means "build the stage 1 compiler and use that to check the target; it's useful when you're using `cfg(bootstrap)` in the standard library or elsewhere.

Reply:
I have updated this part of the patch.